File: backend/SportTeamManager.Domain/Entities/Training.cs
namespace SportTeamManager.Domain.Entities;

public class Training
{
    public Guid Id { get; set; }
    public DateTime Date { get; set; }
    public TimeSpan Duration { get; set; }
    public string Location { get; set; } = string.Empty;
    public string Focus { get; set; } = string.Empty; // Technical, Tactical, Physical, etc.
    public string? Notes { get; set; }
    public Guid TeamId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    // Navigation properties
    public virtual Team Team { get; set; } = null!;
    public virtual ICollection<TrainingAttendance> Attendances { get; set; } = new List<TrainingAttendance>();
}

File: sport-team-manager-frontend/src/app/models/training.ts
import { Player } from "./player";

export interface Training {
  id: string;
  date: string;
  duration: string;
  location: string;
  focus: string;
  notes?: string;
  teamId: string;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateTraining {
  date: string;
  duration: string;
  location: string;
  focus: string;
  notes?: string;
  teamId: string;
}

export interface UpdateTraining {
  date: string;
  duration: string;
  location: string;
  focus: string;
  notes?: string;
}

export interface TrainingWithAttendances extends Training {
  attendances: TrainingAttendance[];
}

export interface TrainingAttendance {
  id: string;
  trainingId: string;
  playerId: string;
  attended: boolean;
  notes?: string;
  rating?: number;
  player?: Player;
}